### DISCUSSION TITLE
Added support for Anthropic’s Claude 3

### DISCUSSION THREAD

**Discussion by keithschacht** [AllYourBot/hostedgpt#191] [Announcements]
I’d love to have you try out the new support for Claude 3. The most important question of all: is Claude 3 Opus better than GPT-4? I’ve seen multiple reports from on twitter claiming they’ve switched to it so you may have just found yourself an upgraded chatbot to help you be all that much better of an engineer. ;)

But in all seriousness, please let me know if it’s a smooth deploy to your instance and if you run into any issues while using it. I’m hoping that this proves to be the first truly useful “bonus” feature that HostedGPT has which ChatGPT does not. I’ve even been thinking about a “dual bot” mode where every question you ask gets replied to by GPT-4 and Claude 3 and maybe you have a left-right arrow to see switch between them or you can hover over the message to see the alternate. One bot will start off as the default but at any point you could decide the other one is better and flip the default for this whole conversation to your preferred bot.

cc: @robacarp @florianfelsing @anandvc @olimart

**Comment by olimart** [comment]
I think support for multiple AI is key since things move (too) fast, so the ability to pick & choose is great (at some point some AI will become obsolete too)
Personally, support for Mistral is what I'm looking for. Haven't had the chance to try Claude yet.

The dual bot option is interesting but instead of running side-by-side (x2 in terms of $ price for each query), I would introduce the ability to "transfer request/discussion" to an alternate AI in order to eventually get a better answer. This way, it only runs when you choose so.

Technically speaking, this means "user keys" must be flexible to quickly add support for new AI. Also, on top of keys, some could run at a custom URL.

**Reply by keithschacht** [reply]
Yes, in addition, I don’t want to lose my conversation history as I move to a new AI and I don’t want to learn the nuances of a new UI. I like the idea of a front-end that is independent of any one AI.

I really like this “transfer discussion” idea! It could actually be integrated into the re-generate button. Maybe that icon becomes something different, and clicking it triggers a drop-down with the default option to “Re-generate” but additional options to “Try with ____” for each of the assistants you have set up.

What makes Mistral the one you’re looking for? I’m curious.

Also, it looks like Mistral has a REST API as well: https://docs.mistral.ai/api  You could follow the PR that I just did for Claude to get this done pretty easily. Now that I worked through a little bit of the pattern, it probably wouldn’t take much more than an hour.

**Comment by olimart** [comment]
> What makes Mistral the one you’re looking for? I’m curious.

It seems to better understand other languages (could be wrong though).

**Comment by florianfelsing** [comment]
Exciting news!

I'll give it a spin during the day and then report back.

**Comment by keithschacht** [comment]
Already, next up is to replicate all the features of Devin within HostedGPT:
https://twitter.com/cognition_labs/status/1767548763134964000

Next week I'm going to finish up the ability to speak to HostedGPT. After that, I'll get started on the Devin features!

**Reply by anandvc** [reply]
I want HostedDevin! 😄

**Comment by keithschacht** [comment]
@olimart I just implemented the "change assistant" idea you suggested. It feels really nice! It's really easy to compare assistants now!

The way it works is whenever you are having a conversation with an assistant, there is always a re-generate icon beneath all the assistant's replies. If you click that icon, it reveals a drop-down menu showing you all your assistants. The first assistant in that list is always the one you're currently using — in case you simply want to re-generate the "normal" way. But alternatively, you can select a different assistant to re-generate with. When you do that, all subsequent messages in this conversation are with that assistant (unless you repeat and change it back).

I was was asking a series of hard technical questions to GPT-4. It didn't know one of them even after I asked it twice so I re-generated the second failure and asked Claude 3 instead. He got it right! It's elegant to selectively switch assistants mid-conversation.

<img src="https://p425.p0.n0.cdn.zight.com/items/ApuPWXb6/c413f8dc-519f-4f5f-b250-45a7688b99d1.gif?v=11dfe0c7469151b6b06b46cb8e2c1e29">

cc: @anandvc since I know you want to try Claude 3

**Reply by olimart** [reply]
Nicely done @krschacht . Thanks for implementing this.
